File: tests/cs-sdk/unit/nogui_opensocket_responser_test.cs
// unit test: OpenSocketResponser (nogui)
using System;
using System.Threading;
using System.Threading.Tasks;
using KnotLink;

var res = new OpenSocketResponser("app.knotlinksdk.test", "test_socket")
{
    OnQuestionAsync = data =>
    {
        Console.WriteLine($"Received: {data}");
        return Task.FromResult($"Echo: {data}");
    }
};
Console.WriteLine("Responser running...");
Thread.Sleep(60000);
res.Dispose();

File: knotlink-sdk-cpp/tests/unit/nogui_opensocket_responser_test.cpp
// unit test: OpenSocketResponser (nogui)
#include "../../knotlink/OpenSocketResponser.hpp"
#include <iostream>
#include <thread>
#include <chrono>
using namespace knotlink;

int main() {
    OpenSocketResponser res("app.knotlinksdk.test", "test_socket");
    res.setQuestionHandler([](const std::string& data) -> std::string {
        std::cout << "Received: " << data << std::endl;
        return "Echo: " + data;
    });
    std::cout << "Responser running..." << std::endl;
    std::this_thread::sleep_for(std::chrono::seconds(60));
    return 0;
}
